packages/core/src/App/Containers/RealAccountSignup/__tests__/account-wizard.spec.tsx

```diff
@@ -185,4 +205,13 @@ describe('<AccountWizard />', () => {
         expect(mock_props.fetchStatesList).toBeCalledTimes(1);
         expect(screen.getByText('TestComponent')).toBeInTheDocument();
     });
+
+    it('should invoke Create account and IDV data submission APIs on click of Submit button', async () => {
+        render(<AccountWizard {...mock_props} />);
+        const ele_submit_btn = screen.getByRole('button', { name: 'Submit' });
+        await waitFor(() => {
```

Comment:
is it expected to wrapp fireEvent with waitFor?
I think the expectation can be wrapped with waitFor.
Could you please try?
@likhith-deriv

Reply:
Assertion wrapped inside waitfor callback will pass for both +ve and -ve scenarios. I have tried this